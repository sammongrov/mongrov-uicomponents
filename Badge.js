import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { styles } from 'react-native-theme';
import PropTypes from 'prop-types';

export default class Badge extends Component {
  render() {
    const { messageCount } = this.props;
    return (
      <View style={styles.badgeContainer}>
        <Text style={styles.badgeText}>{messageCount < 99 ? messageCount : '99+'}</Text>
      </View>
    );
  }
}

Badge.propTypes = {
  messageCount: PropTypes.number.isRequired,
};

Badge.defaultProps = {};
