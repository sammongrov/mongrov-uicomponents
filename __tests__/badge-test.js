import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Alert } from 'react-native';
import Badge from '../Badge';

configure({ adapter: new Adapter() });
const messageCount = () => Alert.alert('message');

it('Badge renders correctly', () => {
  const tree = renderer.create(<Badge messageCount={12} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('messageCount Component using shallow renders correctly', () => {
  const title = 'message';
  const wrapper = shallow(<Badge title={title} messageCount={messageCount} />);
  expect(wrapper.instance().props.messageCount).toBeTruthy();
});
