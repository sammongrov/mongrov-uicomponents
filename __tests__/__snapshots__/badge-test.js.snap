// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    12
  </Text>
</View>
`;
